data_model/src/events/time.rs

```diff
@@ -0,0 +1,110 @@
+//! Time event and filter
+
+use core::{ops::Range, time::Duration};
+
+use super::*;
+
+/// Special event that is emitted when `WSV` is ready for handling time-triggers
+///
+/// Contains time interval which is used to identify time-triggers to be executed
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Decode, Encode, IntoSchema)]
+pub struct Event(pub Interval);
```

Comment:
The basic idea is to emmit `TimeEvent` to find all the time-based triggers that should be executed at given time interval. Interval always starts from block timestamp and has length == commit max time

Reply:
If we have only *start* point, that's unclear how to find proper triggers

Reply:
I don't see why seconds appear in our discussion. Our time resolution is the block commit for now

Reply:
Do we have multiple timed-trigger executions within a round then?

Reply:
If you are talking about one trigger to be executed multiple times in one round -- then **no**, it will be executed only once. But may be we should execute it mutiple times?
If a trigger should be executed every 1 minute, and commitment time is 5 minute, then trigger should be executed 5 times, not one... Makes sesne

Reply:
After a meeting we (@Arjentix , @s8sato and @appetrosyan) have decided to use the current implementation with additional changes (see new commits). We are not using *clock tick* events.

Triggers, that should be executed more than once in estimated time will be executed multiple times. My new realization also watches if a new block commit has occurred before or later estimated time and corrects the numbers of executions based on previous estimating. Consider the scheme below:

```
----|----*----*!---*--|-*----*---!*-
    b1        e1      b2        e2

| -- block
* -- point, when trigger ideally should be executed
! -- estimation of the next block commit

b1 -- block 1 commit
b2 -- block 2 commit
e1 -- estimation for b2
e2 -- estimation for b3
```

|Phase|Description|
|-----|-----------|
|`b1` commit|Trigger will be executed **2** times, cause before `e1` there are only two `*`|
|`b2` commit|Trigger will be executed **3** times, cause there are two `*` between `b2` and `e2` and also there is one in the gap between `e1` and `b2`|